### PR TITLE
Update release yaml to include manual validation jobs

### PR DESCRIPTION
**Pull request by alerickson**
# PR Summary
Add manual validation before publishing to PowerShell internal feed and before publishing to PowerShell Gallery.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

File: .ci/release.yml [hunk 1]
```diff
@@ -1,11 +1,30 @@
...
+    displayName: ${{ parameters.displayValidationJob1 }}
+    pool: server
+    timeoutInMinutes: 4320 # job times out in 3 days
+    steps:
+    - task: ManualValidation@0
+      displayName: ${{ parameters.displayValidationJob1 }}
+      timeoutInMinutes: 2880 # 2 days
+      inputs:
+        instructions: ${{ parameters.instructionsValidationJob1 }}
+        onTimeout: reject
```
Comment (adityapatwardhan) [inline .ci/release.yml]:
```suggestion
- job: ${{ parameter.validationJob1 }}
  displayName: ${{ parameters.displayValidationJob1 }}
  pool: server
  timeoutInMinutes: 4320 # job times out in 3 days
  steps:
  - task: ManualValidation@0
    displayName: ${{ parameters.displayValidationJob1 }}
    timeoutInMinutes: 2880 # 2 days
    inputs:
      instructions: ${{ parameters.instructionsValidationJob1 }}
      onTimeout: reject
```

File: .ci/release.yml [hunk 2]
```diff
@@ -47,8 +59,20 @@ jobs:
       nuGetFeedType: external
       publishFeedCredentials: PSCoreReleasePush
 
+- job: ${{ parameter.validationJob2 }}
```
Comment (adityapatwardhan) [inline .ci/release.yml:62]: this is missing the depends on ${{ parameters.jobName1 }}

File: .ci/release.yml [hunk 3]
```diff
@@ -47,8 +59,20 @@ jobs:
...
+    pool: server
+    timeoutInMinutes: 4320 # job times out in 3 days
+    steps:
+    - task: ManualValidation@0
+      displayName: ${{ parameters.displayValidationJob2 }}
+      timeoutInMinutes: 2880 # 2 days
+      inputs:
+        instructions: ${{ parameters.instructionsValidationJob2 }}
+        onTimeout: reject
+
```
Comment (adityapatwardhan) [inline .ci/release.yml]:
```suggestion
- job: ${{ parameter.validationJob2 }}
  displayName: ${{ parameters.displayValidationJob2 }}
  pool: server
  timeoutInMinutes: 4320 # job times out in 3 days
  steps:
  - task: ManualValidation@0
    displayName: ${{ parameters.displayValidationJob2 }}
    timeoutInMinutes: 2880 # 2 days
    inputs:
      instructions: ${{ parameters.instructionsValidationJob2 }}
      onTimeout: reject

```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # CHANGELOG
-## 0.9.0-rc1
```
Comment (adityapatwardhan) [inline CHANGELOG.md]: unrelated changes?

File: src/Microsoft.PowerShell.PSResourceGet.psd1 [hunk 5]
```diff
@@ -45,7 +45,7 @@
         'udres')
     PrivateData = @{
         PSData = @{
-            Prerelease   = 'rc1'
+            Prerelease   = 'rc'
```
Comment (adityapatwardhan) [inline src/Microsoft.PowerShell.PSResourceGet.psd1]: unrelated changes?

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: .ci/ci_release.yml [hunk 6]
```diff
@@ -292,7 +292,7 @@ stages:
           BuildDropPath: $(signOutPath)
           Build_Repository_Uri: 'https://github.com/powershell/PSResourceGet'
           PackageName: 'Microsoft.PowerShell.PSResourceGet'
-          PackageVersion: '0.9.0-rc1'
+          PackageVersion: '0.9.0-rc'
```
Comment (alerickson) [inline .ci/ci_release.yml]:
```suggestion
          PackageVersion: '0.9.0-rc1'
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,5 @@
...
+## 0.9.0-rc
```
Comment (alerickson) [inline CHANGELOG.md]:
```suggestion
## 0.9.0-rc1
```

File: src/Microsoft.PowerShell.PSResourceGet.psd1 (hunk 5 above)
Comment (alerickson) [inline src/Microsoft.PowerShell.PSResourceGet.psd1]:
```suggestion
            Prerelease   = 'rc1'
```

File: src/Microsoft.PowerShell.PSResourceGet.psd1 [hunk 8]
```diff
@@ -55,7 +55,7 @@
             ProjectUri   = 'https://go.microsoft.com/fwlink/?LinkId=828955'
             LicenseUri   = 'https://go.microsoft.com/fwlink/?LinkId=829061'
             ReleaseNotes = @'
-## 0.9.0-rc1
+## 0.9.0-rc
```
Comment (alerickson) [inline src/Microsoft.PowerShell.PSResourceGet.psd1]:
```suggestion
## 0.9.0-rc1
```

File: .ci/release.yml [hunk 9]
```diff
@@ -47,8 +59,20 @@ jobs:
       nuGetFeedType: external
       publishFeedCredentials: PSCoreReleasePush
 
+- job: ${{ parameter.validationJob2 }}
+  displayName: ${{ parameters.displayValidationJob2 }}
```
Comment (alerickson) [inline .ci/release.yml:64]:
```suggestion
  dependsOn: ${{ parameters.jobName1 }}
  displayName: ${{ parameters.displayValidationJob2 }}
```

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**